homeassistant/components/twinkly/__init__.py

```diff
@@ -33,9 +33,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except (asyncio.TimeoutError, ClientError) as exception:
         raise ConfigEntryNotReady from exception
 
+    software_version = await client.get_firmware_version()
+
     hass.data[DOMAIN][entry.entry_id][DATA_CLIENT] = client
     hass.data[DOMAIN][entry.entry_id][DATA_DEVICE_INFO] = device_info
-
+    hass.data[DOMAIN][entry.entry_id][ATTR_SW_VERSION] = software_version.get(
+        ATTR_VERSION
+    )
```

Comment:
It would be better to construct the dict and store it via something like so you don't have to do multiple dict lookups to set each key value pair

``` 
hass.data[DOMAIN][entry.entry_id] = {
    DATA_CLIENT: client,
....
}
```

Reply:
Consider storing data in a dataclass instead of a `dict` in a future PR

example
https://github.com/home-assistant/core/blob/d4a2927ebe28c1309250058f0c194ff69419b220/homeassistant/components/lookin/models.py#L19

Reply:
Thanks.  I'll get back to that later.